### PR TITLE
Fixes several demos

### PR DESCRIPTION
**Pull request by rrahn**
Fixes build errors and warnings for some demos.

### CODE REVIEW

**Comment by rrahn** [comment]
Note that files that were prefixed with UNUSED or were located in the unused or unknown section have not been updated as long as it is unclear whether to keep them or to delete them after all.

**Comment by esiragusa** [comment]
Can't we delete them?

**Comment by rrahn** [comment]
Well, I believe so, if they are not used and no one objects?

**Comment by rrahn** [comment]
but I think it is easier to merge this PR and delete the files in a different pr. Then I don't have to merge master and develop branches again, but can apply a standard hotfix.

**Comment by esiragusa** [comment]
Yes sure :+1:

**Comment by rrahn** [comment]
Please do not merge yet. 
Waiting for #1226 to be merged and then rebased onto updated master.

**Comment by rrahn** [comment]
@h-2  can you please check this PR and then merge if everything is fine (don't forget #1229)

**Comment by h-2** [comment]
@rrhan will do

File: demos/dox/sequence/segment.cpp [hunk 1]
```diff
@@ -94,5 +94,27 @@ int main()
...
+    ignoreUnusedVariableWarning(_inf2);
+    ignoreUnusedVariableWarning(_inf3);
+    ignoreUnusedVariableWarning(_inf4);
+    ignoreUnusedVariableWarning(_inf5);
+    ignoreUnusedVariableWarning(_inf6);
+    ignoreUnusedVariableWarning(_inf7);
+    ignoreUnusedVariableWarning(_inf8);
+    ignoreUnusedVariableWarning(_pref2);
+    ignoreUnusedVariableWarning(_suf2);
+
```
Comment (h-2) [inline demos/dox/sequence/segment.cpp]: Is there no nicer way to do this? e.g. include the SEQAN_UNUSED macro in the typedef definition?

**Pull request merged**